### ISSUE TITLE
TLU inplace

### ISSUE THREAD

**Issue by qubvel** [tattaka/Filter-Response-Normalization-PyTorch#1]
Hi, thanks for FRN and TLU implementations
I have doubds that is an inplace version of TLU
```python
if self.inplace:
        input = torch.max(input, self.tau)
        return input
```
https://github.com/tattaka/Filter-Response-Normalization-PyTorch/blob/7dc4e508c9984f8ef2cf3efb0e0ed899fcff6757/filter_response_normalization/filter_response_normalize.py#L144

**Comment by tattaka** [comment]
Thank you for pointing out the mistake!
In the latest version of the master branch I fixed it as below.
```python
if self.inplace:
    return torch.max(input, self.tau)
else:
    input = torch.max(input, self.tau)
    return input
```

https://github.com/tattaka/Filter-Response-Normalization-PyTorch/blob/fba5f5eb0a159eb8f6895240b2c61e695d4e85dc/filter_response_normalization/filter_response_normalize.py#L144

Your `segmentation_models.pytorch` repository helped me a lot in the cloud competition on Kaggle.
Thank you for your great contribution.

**Comment by vadimkantorov** [comment]
@tattaka Tensors are returned by reference in PyTorch, so your fix won't achieve inplace. You probably want to use `clamp_` or some other `*_` version to preserve autograd (`out=` version isn't currently supported by pytorch autograd as far as I know)

**Comment by tattaka** [comment]
@vadimkantorov If `clamp_` is used,` max` and `min` cannot receive a tensor and cannot be implemented.
According to [here](https://github.com/pytorch/pytorch/issues/1641), it seems to work by using `out=`. 
If you have other good ideas, it will be helpful if you suggest them.

**Comment by vadimkantorov** [comment]
A problem is that out tensor wouldn't have grad_fn set and won't be attached to autograd graph, if I understand the current PyTorch mechanics.

**Comment by tattaka** [comment]
Thank you, I understand this issue. Certainly `out =` doesn't seem to support autograd.
I think of another solution.

**Comment by qubvel** [comment]
Did you try `torch.threshold_`?

**Comment by tattaka** [comment]
Since position 2 and 3 of `torch.threshold_` augment were value, I thought that this could not be implemented. (I am a beginner of pytorch and may not fully understand how to use it...)

**Comment by tattaka** [comment]
@qubvel @vadimkantorov 
Since it seems difficult to implement the `inplace` version under the current situation, I will delete the option once.
Thank you for the great knowledge. 🙏

**Comment by qubvel** [comment]
I have also suggestion about activated norm (include activation optionally). This will help to convert models - replace batchnorm with activated frn and replace relu with identity module.

P.S. look at #2
